docs/authentication/saml/custom-provider.mdx

```diff
@@ -0,0 +1,123 @@
+description: Learn how to integrate an Identity Provider with Clerk using SAML SSO.
+---
+
+# Custom provider
+
+<TutorialHero
+  beforeYouStart={[
+    {
+      title: "Add the Enhanced authentication add-on to your Pro plan",
+      link: "https://clerk.com/pricing"
+    },
+    {
+      title: "Enable email address as an identifier for your application.",
+      link: "/docs/authentication/configuration/sign-up-sign-in-options#identifiers",
+    }
+  ]}
+>
+
+- Use a custom IdP to enable single sign-on (SSO) via SAML for your Clerk application.
+
+</TutorialHero>
+
+Clerk supports Enterprise SSO via the SAML protocol, enabling you to create authentication strategies for an Identity Provider (IdP). Currently, Clerk offers direct integrations with [Microsoft Azure AD](/docs/authentication/saml/azure), [Google Workspace](/docs/authentication/saml/google), and [Okta Workforce](/docs/authentication/saml/okta) as IdPs. However, you can also integrate with any other IdP that supports the SAML protocol. This guide will show you how to set up a SAML connection with a custom IdP in Clerk.
+
+## Tutorial
+
+<Steps>
+
+### Set up an enterprise connection in Clerk
+
+To create a SAML connection in Clerk:
+
+1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com).
+2. In the navigation sidebar, select **User & Authentication > [Enterprise Connections](https://dashboard.clerk.com/last-active?path=user-authentication/enterprise-connections)**.
+3. Select the **Create connection** button.
+4. You will be presented with a modal to create a new connection. Fill in the required fields and for the **Identity Provider**, select **Custom**.
+5. Next, on the **Connection** page, **do not select the Enable connection toggle just yet.** You need to fully configure your IdP first before exposing your connection to your users.
+6. Leave this page open. You will need to come back to it to complete the setup.
+
+### Create a new enterprise application in your IdP
+
+Create a new application in your IdP. In the next steps, you will configure your IdP with the settings provided by your service provider (Clerk), and you will configure Clerk with the settings provided by your IdP. So, you will need to keep both the IdP and Clerk dashboards open.
+
+### Configure your service provider
+
+To configure your service provider (Clerk), your IdP will either ask for the **Assertion Consumer Service (ACS) URL** and **Entity ID** or it will ask for the **Metadata URL**. If your IdP gives you the option to choose between the two, it is recommended to choose the **Metadata URL** as it is the quickest and most reliable way to configure your service provider.
+
+Here are what these settings mean:
+
+- **Assertion Consumer Service (ACS) URL** - This is a unique identifier for your SAML connection that your IdP application needs.
+- **Entity ID** - This is your application's URL that your IdP will redirect your users back to after they have authenticated in your IdP.
```

Comment:
Need to reverse these two. 
ACS URL is the application url where the user will be redirected
Entity ID is the unique identifier